include/dlaf/permutations/general.h

```diff
@@ -109,15 +110,16 @@ void permute(SizeType i_begin, SizeType i_end, Matrix<const SizeType, D>& perms,
 /// @pre @p mat_out has blocksize (NB x NB)
 /// @pre @p mat_out has tilesize (NB x NB)
 ///
-/// Note: The Pipeline<> API allows to use permute() within other algorithms without having to clone communicators
-///       internally.
+/// Note: The CommunicatorPipeline API allows to use permute() within other algorithms without having to
+///       clone communicators internally.
```

Comment:
This comment here caught my eye, because it was not straightforward for me remember how this algorithm worked.

I did some checks, and with the help of @rasolca, we realised that the implementation fo this algorithm actually need some cleanup too. See the comment in the implementation for reference, and in case, we can remove this note here.

Reply:
I changed the implementtion, but I'm not sure the comment is inaccurate even with the changed implementation (unless I'm misunderstanding what it refers to). I'm not sure the comment adds much either so I'd be fine to remove it as well. What would you prefer?

Reply:
I'm not sure, but I would remove it in any case. I mean, otherwise we should write this to every algorithm.

With your work in this PR, I think it should be "clear" from the arguments what guarantees a function gives or not. Isn't it?

Reply:
Yep, I agree!

Reply:
https://github.com/eth-cscs/DLA-Future/pull/993/commits/286628ba436c1b00e295cf4254af92034a81ee4f